contracts/savings-app/src/handlers/execute.rs

```diff
@@ -169,11 +173,17 @@ fn withdraw(
         .add_message(withdraw_msg))
 }
 
-fn autocompound(deps: DepsMut, env: Env, _info: MessageInfo, app: App) -> AppResult {
-    // TODO: shouldn't we have some limit either:
-    // - config.cooldown
-    // - min rewards to autocompound
+fn autocompound(deps: DepsMut, env: Env, info: MessageInfo, app: App) -> AppResult {
     // Everyone can autocompound
+    let config = CONFIG.load(deps.storage)?;
+    let status = get_position_status(
+        deps.storage,
+        &env,
+        config.autocompound_cooldown_seconds.u64(),
+    )?;
+    if !status.is_ready() {
```

Comment:
Do we want to enforce that it needs to wait until after the cooldown? Imo there's no disadvantage to allowing it to be performed earlier.

Reply:
Pretty sure yes, because it's one of the way for user to limit how much gas coins reward user forwards to the autocompounder executor. Do you think this should only be checked when called by non-admin?

Reply:
Or we can only return reward if the cooldown is over. If any earlier calls are made there's no reward?

Reply:
Fixed